### DISCUSSION TITLE
On customisation

### DISCUSSION THREAD

**Discussion by Secret-chest** [linuxmint/discussions#581] [Ideas]
I don't get Clem's intention to remove options ("tuning").

You aren't *asked* to change it, you are *allowed* to. And even though it's not instantly visible in a screenshot on reddit, there are things you use every day that should be customisable. For example the volume step; that's been discussed a lot, but rejected. I don't understand. It's not the most impactful change to the look of your desktop, but think how often you press the volume buttons. It's a small change that makes it much more pleasant. And you can't tune it for everyone, since everyone has different needs and tastes. Same with animations and all.

**Comment by JosephMcc** [comment]
> I don't get Clem's intention to remove options ("tuning").

Can you give some examples of things that clem "intends" on removing? Neither of your examples fits that criteria.

**Reply by Secret-chest** [reply]
Why not?

**Reply by Secret-chest** [reply]
Why are my examples not good?

**Comment by anaximeno** [comment]
Making everything tunable comes with its issues, even small changes can have big effects on systems, because when lots of things depend on lots of things on the system, a small change can lead to great complications, so every change should be made carefully enough to not complicate things for future changes (that eventually have to be made to improve the system). Extensions are good ways to extend behavior for those that want more specific behaviors, and at least related to Cinnamon, it's not very hard to modify its components. And I also have the same question as Joseph above.

**Reply by Secret-chest** [reply]
Except when it's currently a constant and you can easily add a way to set it to a custom value. The plumbing is there.

**Reply by anaximeno** [reply]
You can change the sound controller from an extension or create an extension to control the surround as was suggested by the sound applet. Not all things must be in the default experience. I like KDE but having many settings made me not use it in the past because it was a bit too confusing having all these settings when I just wanted to make a simple change. GNOME is criticized for keeping things with few ways to change ootb which I understand to some extent (even though you can use lots of extensions with GNOME as well). The ideal is to have a good balance, I think Cinnamon maintains a good balance.

**Reply by Leokuma** [reply]
As long as I'm able to customize even the most ridiculous things, I don't mind if it's via extensions or built into Cinnamon.

**Comment by Leokuma** [comment]
Answering the questions above, Clem left the following message on [this PR:](https://github.com/linuxmint/cinnamon/pull/8884#issuecomment-553994539)

> We're removing similar features elsewhere in Cinnamon, for instance asking the user to fine tune the number of millisecond a particular animation effect takes.

That was years ago, though. Cinnamon is still highly customizable.

I don't understand the inner workings and architecture of Cinnamon, so I'm unable to judge which features would be a maintenance burden.

Anyway, as a user, I do wish that the maintainers keep Cinnamon ridiculously customizable as long as it doesn't become unmaintainable. Customization is Cinnamon's main feature, it's what makes it unique. All the other desktops are heading towards a "fixed" UX, similar to Apple's philosophy. Mac was never customizable; Windows is getting less and less customizable. Cinnamon is filling that gap.

**Reply by Secret-chest** [reply]
The animation system still allows to change them IIRC, they're just hardcoded now.

**Comment by Jopp-gh** [comment]
I didn't update to Wilma but think that Mint Virginia is pretty customizable, sometimes even too much (amount of installed themes, many of them aren't perfectly integrated to include all features, not even Mint-y - so lets reduce them to just a hand of themes in total). The key here would be to download  theme *resources* (like mime types, emblems, etc) from https://cinnamon-spices.linuxmint.com/

Some other features are still missing in Nemo (to get selected files/ folders also outside context menus) 

Lately, I've added some idea posts and opinions to other people's posts here on this discussion page, how I'd improve some aspects of Mint. 

For instance, the **installation process** here:
https://github.com/orgs/linuxmint/discussions/590#discussioncomment-10474623

**Improving Nemo** here:
https://github.com/orgs/linuxmint/discussions/583#discussioncomment-10412776
And here:
https://github.com/orgs/linuxmint/discussions/584

**Notifications**:
https://github.com/orgs/linuxmint/discussions/489#discussioncomment-10441584

**Mint's search panel**
https://github.com/orgs/linuxmint/discussions/592

**Comment by Secret-chest** [comment]
Animations are hardcoded. But maybe allowing extensions to mod them would be even better.
